paging_test.py

```diff
@@ -3423,19 +3423,31 @@ def test_failure_threshold_deletions(self):
         supports_v5_protocol = self.supports_v5_protocol(self.cluster.version())
 
         self.fixture_dtest_setup.allow_log_errors = True
+        supports_guardrails = self.cluster.version() >= LooseVersion('4.0')
+        if supports_guardrails:
+            config_opts = {'guardrails': {'tombstone_failure_threshold': 500,
+                                          'tombstone_warn_threshold': -1,
+                                          'write_consistency_levels_disallowed': {}},
+                           'tombstone_failure_threshold': 0,
+                           'tombstone_warn_threshold': 0}
+        else:
+            config_opts = {'tombstone_failure_threshold': 500}
         self.cluster.set_configuration_options(
             values={'tombstone_failure_threshold': 500}
         )
         self.session = self.prepare()
         self.setup_data()
 
-        # Add more data
+        if supports_guardrails:
```

Comment:
I don't see relevance of that change wrt the expected scenario.
Do guardrails change how do we treat tombstones?
At least it requires some more detailed explanation

Reply:
Hmm, not sure.
@Gerrrr maybe you would be able to comment?

Reply:
I think the reason is in the following comment
```
            # cell tombstones are not counted towards the threshold, so we delete rows
```
So it seems deleting a cell does not count toward the tombstone threshold with guardrails, so the change is to delete the row instead of using `null` to create a tombstone cell.

Reply:
you mean - guardrails not only add guardrails but also change how we treat tombstones? hmmm

Reply:
It affects how tombstones are counted for guardrail thresholds - https://datastax.jira.com/browse/DB-2426